Yosemite/Yosemite/Stores/ProductStore.swift

```diff
@@ -351,6 +374,41 @@ private extension ProductStore {
 }
 
 
+// MARK: - Storage: Search Results
+//
+private extension ProductStore {
+
+    /// Upserts the Products, and associates them to the SearchResults Entity (in Background)
+    ///
+    private func upsertSearchResultsInBackground(siteID: Int, keyword: String, readOnlyProducts: [Networking.Product], onCompletion: @escaping () -> Void) {
+        let derivedStorage = sharedDerivedStorage
+        derivedStorage.perform { [weak self] in
```

Comment:
Remove `[weak self]`. The caller is already set to `[weak self]`, in [L74](https://github.com/woocommerce/woocommerce-ios/pull/1289/files#diff-124a45ac85ac99b415c1c94e3e7eaedaR74).

Reply:
from my understanding, the closure here has a different scope and owner from in [L74](https://github.com/woocommerce/woocommerce-ios/pull/1289/files#diff-124a45ac85ac99b415c1c94e3e7eaedaR74) - in L74, the owner of the closure is `remote.searchProducts` while that of the following closure is `derivedStorage.perform`. I generally add `[weak self]` for most of the closures just to be on the safe side. Confirming with @ctarda